agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4706,6 +4708,28 @@ bool slave_thread::handle_client_capability_query(Socket *sd, ieee1905_1::CmduMe
     return true;
 }
 
+bool slave_thread::handle_ap_metrics_query(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    
+    auto tlv_type = cmdu_rx.getNextTlvType();
+    while (tlv_type != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
```

Comment:
good point , fixed .

Reply:
After some digging, I see that I was wrong. Apparently, every CMDU could have VS_TLV inside it, and it can be first.
It means that you must use the while loop as you did, and add a handler for VS_TLV.

Reply:
Since Tomer said it has a pending solution for it, we can just leave it as is.